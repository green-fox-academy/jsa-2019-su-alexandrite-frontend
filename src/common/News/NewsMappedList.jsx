import React from 'react';
import PropTypes from 'prop-types';
import NewsListItem from './NewsListItem';

const NewsMappedList = ({ news }) => (
  news.map((item, i) => (
    <NewsListItem
      key={item.id}
      item={item}
      style={[i !== news.length - 1 && { marginBottom: 15 }]}
    />
  ))
);

NewsMappedList.propTypes = {
  news: PropTypes.arrayOf(
    NewsListItem.propTypes.item,
  ),
};

NewsMappedList.defaultProps = {
  news: [],
};

export default NewsMappedList;
