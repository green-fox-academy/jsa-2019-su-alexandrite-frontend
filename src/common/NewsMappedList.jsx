import React from 'react';
import PropTypes from 'prop-types';
import NewsCard from './NewsCard';

const NewsMappedList = ({ news }) => (
  news.map((item, i) => (
    <NewsCard
      key={item.url}
      item={item}
      style={[i !== news.length - 1 && { marginBottom: 15 }]}
    />
  ))
);

NewsMappedList.propTypes = {
  news: PropTypes.arrayOf(
    NewsCard.propTypes.item,
  ),
};

NewsMappedList.defaultProps = {
  news: [],
};

export default NewsMappedList;
