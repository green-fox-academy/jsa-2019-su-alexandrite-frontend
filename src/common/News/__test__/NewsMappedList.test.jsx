import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from 'react-navigation-hooks';
import NewsMappedList from '../NewsMappedList';

jest.mock('react-navigation-hooks');
jest.mock('../useFetchStockNews');

const sampleNews = [
  {
    id: '123',
    datetime: '2020-01-10T04:46:05.638Z',
    headline: 'headline',
    url: 'http://test.test',
    image: 'http://test.test/test.jpg',
  },
  {
    id: '321',
    datetime: '2020-01-10T04:46:05.638Z',
    headline: 'headline',
    url: 'http://test.test',
    image: 'http://test.test/test.jpg',
  },
];

describe('<NewsMappedList />', () => {
  beforeEach(() => {
    useNavigation.mockReturnValue({
      push: jest.fn(),
      navigate: jest.fn(),
    });
  });

  afterAll(() => {
    jest.resetModules();
  });

  it('Should render properly and match the snapshot', () => {
    const tree = renderer.create(
      <NewsMappedList news={sampleNews} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
