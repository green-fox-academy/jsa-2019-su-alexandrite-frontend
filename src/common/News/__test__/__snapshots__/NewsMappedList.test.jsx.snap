// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsMappedList /> Should render properly and match the snapshot 1`] = `
Array [
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      style={
        Object {
          "borderRadius": 6,
          "marginBottom": 15,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "#ccc",
            "borderRadius": 6,
            "height": 169,
            "overflow": "hidden",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "http://test.test/test.jpg",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 169,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                0,
                0,
                2281701376,
                4278190080,
              ],
              "endPoint": undefined,
              "locations": undefined,
              "startPoint": undefined,
            }
          }
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 12,
              }
            }
          >
            1/10/2020, 12:46:05 PM
          </Text>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 14,
                "fontWeight": "bold",
                "lineHeight": 20,
              }
            }
          >
            headline
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </View>
  </RNGestureHandlerButton>,
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      style={
        Object {
          "borderRadius": 6,
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "backgroundColor": "#ccc",
            "borderRadius": 6,
            "height": 169,
            "overflow": "hidden",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "http://test.test/test.jpg",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 169,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
        <ViewManagerAdapter_ExpoLinearGradient
          proxiedProperties={
            Object {
              "colors": Array [
                0,
                0,
                2281701376,
                4278190080,
              ],
              "endPoint": undefined,
              "locations": undefined,
              "startPoint": undefined,
            }
          }
          style={
            Object {
              "flex": 1,
              "justifyContent": "flex-end",
              "padding": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 12,
              }
            }
          >
            1/10/2020, 12:46:05 PM
          </Text>
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 14,
                "fontWeight": "bold",
                "lineHeight": 20,
              }
            }
          >
            headline
          </Text>
        </ViewManagerAdapter_ExpoLinearGradient>
      </View>
    </View>
  </RNGestureHandlerButton>,
]
`;
